### ISSUE TITLE
[Bug]: LSP reports @typescript-eslint/no-unused-vars, but CLI passes on the same file

### ISSUE THREAD

**Issue by chenjiahan** [web-infra-dev/rslint#671]
### System Info

```text
System:
  OS: macOS 26.2
  CPU: (16) arm64 Apple M3 Max
  Memory: 5.94 GB / 64.00 GB
  Shell: 5.9 - /bin/zsh
Binaries:
  Node: 24.12.0
  npm: 11.6.2
  pnpm: 10.33.0
  bun: 1.3.4
  Deno: 2.5.4
Browsers:
  Chrome: 147.0.7727.56
  Safari: 26.2
npmPackages:
  @rslint/core: ^0.4.2 => 0.4.2
```

Extra editor info:
- VS Code: 1.115.0
- `rstack.rslint`: 0.4.2
- `typescript` in the project: 6.0.2

### Details

I found an inconsistency between the language server and the CLI for `@typescript-eslint/no-unused-vars`.

Reproduction file:
- https://github.com/rstackjs/create-rstack/blob/main/test/skills.test.ts

This callback in `test/skills.test.ts` triggers `rslint` diagnostics in VS Code:

```ts
const xSync = rs.fn((command: string, args: string[], options: unknown) => {
  return {
    stdout: '',
    stderr: '',
    exitCode: 0,
  };
}) as any;
```

In the editor, `rstack.rslint` reports:
- `[@typescript-eslint/no-unused-vars] 'command' is defined but never used.`
- and the same for `args` / `options`

But the CLI passes on the same file with the same config:

```bash
pnpm lint
# Found 0 errors and 0 warnings ...

pnpm exec rslint test/skills.test.ts
# Found 0 errors and 0 warnings ...
```

The config is very small:

```ts
import { defineConfig, ts } from '@rslint/core';

export default defineConfig([
  { ignores: ['**/dist/**'] },
  ts.configs.recommended,
]);
```

I am not sure whether the CLI or the language server is correct here, but I would expect them to be consistent.

This seems different from #555, because here the main problem is that the language server reports the rule while the CLI does not report anything for the same file.

### Reproduce link

https://github.com/rstackjs/create-rstack

### Reproduce Steps

1. Clone `https://github.com/rstackjs/create-rstack`
2. Run `pnpm install`
3. Open `test/skills.test.ts` in VS Code with the `rstack.rslint` extension enabled.
4. Look at the callback parameters around line 70. The language server reports `@typescript-eslint/no-unused-vars`.
5. Run `pnpm lint`.
6. Run `pnpm exec rslint test/skills.test.ts`.
7. Both CLI commands report `Found 0 errors and 0 warnings`.
